monitoring/lifecycle/dashboard.py

```diff
@@ -0,0 +1,283 @@
+        divdLabel += f',status="{code}"'
+    else:
+        divdLabel += ',status!="2xx"'
+
+    if job is not None:
+        divdLabel += f',job="{job}"'
+        divsLabel += f',job="{job}"'
+
+    if process is not None:
+        divdLabel += f',origin="{process}"'
+        divsLabel += f',origin="{process}"'
+
+    return f'sum(rate(lifecycle_s3_operations{{{divdLabel}}}[$__range]))/sum(rate(lifecycle_s3_operations{{{divsLabel}}}[$__range]))'
+
+
+def s3_request_error_rate(title, process=None, job=None, code=None):
+    return Stat(
+        title=title,
+        dataSource="${DS_PROMETHEUS}",
+        format=UNITS.PERCENT_UNIT,
+        targets=[
+            Target(expr=s3_request_error_rate_expr(process, job, code)),
+        ],
+        thresholds=[
+            Threshold("green", 0, 0.0),
+            Threshold("red", 1, 0.1),
+        ],
+    )
+
+
+def s3_request_error_rates(process=None, job=None):
+    return [
+        s3_request_error_rate( "S3 Error Rate", process=process, job=job),
+        s3_request_error_rate( "3xx Error Rate", process=process, job=job, code=STATUS_CODE_3XX),
+        s3_request_error_rate( "4xx Error Rate", process=process, job=job, code=STATUS_CODE_4XX),
+        s3_request_error_rate( "5xx Error Rate", process=process, job=job, code=STATUS_CODE_5XX),
+    ]
+
+
+def s3_deletion_request_time_series(op):
+    successLabel = f'status="2xx",op="{op}",namespace="{"${namespace}"}",job="{"${job_lifecycle_object_processor}"}"'
+    errorLabel = f'status!="2xx",op="{op}",namespace="{"${namespace}"}",job="{"${job_lifecycle_object_processor}"}"'
+
+    return TimeSeries(
+        title=f'{op} requests',
+        dataSource="${DS_PROMETHEUS}",
+        stacking={
+            "group": "A",
+            "mode": "normal",
+        },
+        targets=[
+            Target(
+                expr=f'sum(rate(lifecycle_s3_operations{{{successLabel}}}[$__range]))',
+                legendFormat="success",
+            ),
+            Target(
+                expr=f'sum(rate(lifecycle_s3_operations{{{errorLabel}}}[$__range]))',
+                legendFormat="error",
+            ),
+        ]
+    )
+
+
+def kafka_topic_messages_in_queue(topic, op):
+    return Graph(
+        title=f'Messages in ${topic}',
+        dataSource="${DS_PROMETHEUS}",
+        target=[
+            Target(
+                expr=f'sum(increase(lifecycle_kafka_publish_success{{op="{op}",namespace="{"${namesapce}"}}[$__range]))',
+            ),
+        ],
+    )
+
+
+def kafka_topic_rate(title, op):
+    return Stat(
+        title=title,
+        dataSource="${DS_PROMETHEUS}",
+        targets=[
+            Target(
+                expr=f'sum(rate(lifecycle_kafka_publish_success{{something="{}",some}}[$__range]))',
+             ),
+        ],
+        thresholds=[
+            Threshold("green", 0, 0.0),
+        ],
+    )
+
+
+up = Stat(
+    title="Up",
+    dataSource="${DS_PROMETHEUS}",
+    reduceCalc="lastNotNull",
```

Comment:
should be 'last' : we always want to display the very last value, esp. if there are no more up pods